lib/galaxy/jobs/__init__.py

```diff
@@ -1416,7 +1416,6 @@ def _finish_dataset(self, output_name, dataset, job, context, final_job_state, r
                 try:
                     # Attempt to short circuit NFS attribute caching
                     os.stat(dataset.dataset.file_name)
-                    os.chown(dataset.dataset.file_name, os.getuid(), -1)
```

Comment:
The -1 should force chown not to touch the gid. So this should not be the source of your problem. Note that also the owner is not changed (other wise root would be needed). The intention of this code block is to deal with problems regarding NFS file systems as indicated by the comment above.

Reply:
yes, you are right. I double checked and indeed I can drop the last patch. Only the first two patches are in the place.